cas-server-webapp-validation/src/main/java/org/jasig/cas/web/AbstractServiceValidateController.java

```diff
@@ -128,13 +151,38 @@ protected Credential getServiceCredentialsFromRequest(final WebApplicationServic
                 verifyRegisteredServiceProperties(registeredService, service);
                 return new HttpBasedServiceCredential(new URL(pgtUrl), registeredService);
             } catch (final Exception e) {
-                logger.error("Error constructing pgtUrl", e);
+                logger.error("Error constructing {}", CasProtocolConstants.PARAMETER_PROXY_CALLBACK_URL, e);
             }
         }
 
         return null;
     }
 
+
+    /**
+     * Validate authentication context pair.
+     *
+     * @param assertion the assertion
+     * @param request   the request
+     * @return the pair
+     */
+    protected Pair<Boolean, Optional<MultifactorAuthenticationProvider>> validateAuthenticationContext(
+            final Assertion assertion, final HttpServletRequest request) {
+        final String requestedContext = request.getParameter(this.authenticationContextParameter);
```

Comment:
So let me summarize to see if I understand you correctly:

1. Clients need to pass a special parameter today to request mfa during validation.
2. You are saying, if a client fails to send it yet the service itself says it needs MFA in the registry, we should use the service config to enforce mfa? 

If so, sure. Much easier. Shall I make the change, or would u rather push that yourself?